airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/serde/AirbyteMessageDeserializer.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol.serde;
+
+import io.airbyte.commons.version.AirbyteVersion;
+
+public interface AirbyteMessageDeserializer<T> {
+
+  T deserialize(final String json);
```

Comment:
This means that the deserializer will fail if provided with an invalid json String? It will look like an issue in the migration but it is an issue with what is provided to it. Could we enforce to have a valid json by passing a JsonNode in the method. It won't avoid all the JsonSerialization issue but it will at least avoid passing invalid Json string.

Reply:
I am pretty flexible on this. The initial implementation was using `JsonNode`, part of what felt weird is that we would double call Jsons to go string -> JsonNode -> AirbyteMessage while we actually wrote the combined operation.
I am not sure the two-step split brings us value in our current implementation.
@benmoriceau, can we leave this for now and see in the PR where I actually replace the current parsing logic?

Reply:
Yes we can leave it for now. It just felt that it was taking the risk of not being explicit about what we are expecting as a string format but the multiple conversion is a fair point as well